src/figuresClasses.ts

```diff
@@ -1,11 +1,81 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: 'triangle' | 'circle' | 'rectangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+
+    const sumAB: number = this.a + this.b;
+    const sumAC: number = this.a + this.c;
+    const sumBC: number = this.b + this.c;
+
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Error');
+    }
+
+    if (this.a >= sumAC || this.b >= sumBC || this.c >= sumAB) {
+      throw new Error('Error1');
+    }
+  }
+
+  getArea(): number {
+    const per: number = (this.a + this.b + this.c) / 2;
+    const square: number =
+      per * (per - this.a) * (per - this.b) * (per - this.c);
+
+    return Math.trunc(Math.sqrt(square) * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  public shape: 'triangle' | 'circle' | 'rectangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public radius: number,
+  ) {
```

Comment:
[CODE KNOWLEDGE] - Similar to the previous comment, the 'shape' property is being manually assigned in the constructor, which is unnecessary when using TypeScript's parameter properties feature.